app/react/Library/helpers/requestState.js

```diff
@@ -50,67 +50,60 @@ export function processQuery(params, globalResources, key) {
   };
 }
 
-export default function requestState(request, globalResources, calculateTableColumns = false) {
+export default function requestState(
+  request,
+  globalResources,
+  options = { calculateTableColumns: false, geolocation: false }
+) {
   const docsQuery = processQuery(request.data, globalResources, 'library');
-  const documentsRequest = request.set(
-    tocGenerationUtils.aggregations(docsQuery, globalResources.settings.collection.toJS())
-  );
 
-  const templatesWithGeolocation = globalResources.templates.find(template =>
-    template.get('properties').find(property => property.get('type') === 'geolocation')
+  let documentsRequest = request.set(
+    tocGenerationUtils.aggregations(docsQuery, globalResources.settings.collection.toJS())
   );
 
-  const markersRequest = templatesWithGeolocation
-    ? api.search(
-        request.set({
-          ...docsQuery,
-          geolocation: true,
-        })
-      )
-    : { rows: [] };
-
-  return Promise.all([api.search(documentsRequest), markersRequest]).then(
-    ([documents, markers]) => {
-      const templates = globalResources.templates.toJS();
-      const filterState = libraryHelpers.URLQueryToState(
-        documentsRequest.data,
-        templates,
-        globalResources.thesauris.toJS(),
-        globalResources.relationTypes.toJS(),
-        request.data.quickLabelThesaurus
-          ? getThesaurusPropertyNames(request.data.quickLabelThesaurus, templates)
-          : []
-      );
+  if (options.geolocation) {
+    documentsRequest = request.set({ ...docsQuery, geolocation: true });
+  }
```

Comment:
I am confused by this: the previous behavior was that, any query, be it Cards or Map or Table, the library -> filters -> documentTypes were the result of `request.set` with the `tocGenerationUtils.aggregations`.  Here, you are changing that behavior for the map view when geolocation is asked.   Please, ensure this does not have adverse effects on the filters and aggregations on the side panel.

Reply:
Indeed, this is changing the values of aggregations when you change to map view, I don't think that is the intended behavior.  The aggregations on the library, in whichever "mode", be it Cards, Map or Table View, should show the same aggregations.  The map should show all the map geolocation values, but the aggregation numbers should not change depending on the tab I am in.  They are just different views of the same collection.

Reply:
Map view well do 2 calls like before: one for aggregations, and one with `geolocation` to get markers.